pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -96,7 +96,19 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			log.Error(err, "failed to get labels from virtualmachine", "subnetPort.Name", subnetPort.Name, "subnetPort.UID", subnetPort.UID)
 			return common.ResultRequeue, err
 		}
-		nsxSubnetPortState, err := r.SubnetPortService.CreateOrUpdateSubnetPort(subnetPort, nsxSubnetPath, "", labels)
+
+		isVmSubnetPort := true
+		if subnetPort.Labels[servicecommon.LabelImageFetcher] == "true" {
+			isVmSubnetPort = false
+			if labels == nil {
+				labels = &map[string]string{}
+			}
+			(*labels)[servicecommon.LabelImageFetcher] = "true"
+		}
+		// specified by user, or default to the node id of the VM the pod runs on
+		contextID := subnetPort.Annotations[servicecommon.AnnotationNodeID]
+
+		nsxSubnetPortState, err := r.SubnetPortService.CreateOrUpdateSubnetPort(subnetPort, nsxSubnetPath, contextID, labels, isVmSubnetPort)
```

Comment:
I think we can move this logic into func `buildSubnetPort` so that we don't need to pass `isVmSubnetPort` to `CreateOrUpdateSubnetPort` and `buildSubnetPort` and the code will looks a little clearer. How do you think?